tests/unit/s2n_security_policies_test.c

```diff
@@ -0,0 +1,229 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "s2n_test.h"
+
+#include "tls/s2n_security_policies.h"
+
+int main(int argc, char **argv)
+{
+    BEGIN_TEST();
+
+    const struct s2n_security_policy *security_policy = NULL;
+
+    /* Test common known good cipher suites for expected configuration */
+    {
+        EXPECT_SUCCESS(s2n_find_security_policy_from_version("default", &security_policy));
+        EXPECT_TRUE(s2n_ecc_is_extension_required(security_policy));
+        EXPECT_FALSE(s2n_pq_kem_is_extension_required(security_policy));
+        EXPECT_NULL(security_policy->kem_preferences->kems);
+        EXPECT_EQUAL(0, security_policy->kem_preferences->count);
+        EXPECT_FALSE(s2n_security_policy_supports_tls13(security_policy));
+
+        security_policy = NULL;
+        EXPECT_SUCCESS(s2n_find_security_policy_from_version("default_tls13", &security_policy));
+        EXPECT_TRUE(s2n_ecc_is_extension_required(security_policy));
+        EXPECT_FALSE(s2n_pq_kem_is_extension_required(security_policy));
+        EXPECT_TRUE(s2n_security_policy_supports_tls13(security_policy));
+        EXPECT_EQUAL(0, security_policy->kem_preferences->kems);
+        EXPECT_NULL(security_policy->kem_preferences->kems);
+
+        security_policy = NULL;
+        EXPECT_SUCCESS(s2n_find_security_policy_from_version("test_all", &security_policy));
+        EXPECT_TRUE(s2n_ecc_is_extension_required(security_policy));
+        EXPECT_TRUE(s2n_pq_kem_is_extension_required(security_policy));
+#if !defined(S2N_NO_PQ)
+        EXPECT_EQUAL(4, security_policy->kem_preferences->count);
+        EXPECT_TRUE(s2n_pq_kem_is_extension_required(security_policy));
+        EXPECT_NOT_NULL(security_policy->kem_preferences->kems);
+        EXPECT_EQUAL(security_policy->kem_preferences->kems, pq_kems_r2r1);
+#else
+        EXPECT_EQUAL(0, security_policy->kem_preferences->count);
+        EXPECT_NULL(security_policy->kem_preferences->kems);
+#endif
+
+        security_policy = NULL;
+        EXPECT_SUCCESS(s2n_find_security_policy_from_version("KMS-TLS-1-0-2018-10", &security_policy));
+        EXPECT_TRUE(s2n_ecc_is_extension_required(security_policy));
+        EXPECT_FALSE(s2n_pq_kem_is_extension_required(security_policy));
+        EXPECT_EQUAL(0, security_policy->kem_preferences->count);
+        EXPECT_NULL(security_policy->kem_preferences->kems);
+
+#if !defined(S2N_NO_PQ)
```

Comment:
Can we combine this with the previous #if to keep all of it in one place?

Reply:
Moving them all to one `#if-else` would mess with the organization (all kms-related secuirty_policies test together) and code-logic, I would prefer it to be kept this way.